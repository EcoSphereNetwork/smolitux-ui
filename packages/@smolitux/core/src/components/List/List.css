/* packages/@smolitux/core/src/components/List/List.css */

/* List Base */
.smolitux-list {
  list-style: none;
  margin: 0;
  padding: 0;
  width: 100%;
}

/* List Variants */
.smolitux-list--ordered {
  list-style: decimal;
  padding-left: 1.5rem;
}

.smolitux-list--unordered {
  list-style: var(--list-marker, disc);
  padding-left: 1.5rem;
}

.smolitux-list--description {
  display: grid;
  grid-template-columns: auto 1fr;
  gap: 0.5rem 1rem;
}

.smolitux-list--custom {
  list-style: none;
  padding-left: 0;
}

.smolitux-list--no-indent {
  padding-left: 0;
}

/* List Sizes */
.smolitux-list--sm .smolitux-list-item {
  padding: 0.375rem 0.5rem;
  font-size: 0.875rem;
}

.smolitux-list--md .smolitux-list-item {
  padding: 0.5rem 0.75rem;
  font-size: 1rem;
}

.smolitux-list--lg .smolitux-list-item {
  padding: 0.75rem 1rem;
  font-size: 1.125rem;
}

/* List Density */
.smolitux-list--compact .smolitux-list-item {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.smolitux-list--comfortable .smolitux-list-item {
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
}

/* List Alignment */
.smolitux-list--align-start .smolitux-list-item {
  text-align: left;
}

.smolitux-list--align-center .smolitux-list-item {
  text-align: center;
}

.smolitux-list--align-end .smolitux-list-item {
  text-align: right;
}

/* List with Dividers */
.smolitux-list--dividers .smolitux-list-item--divider {
  border-bottom: 1px solid var(--list-divider-color, #e5e7eb);
}

.smolitux-list--dividers .smolitux-list-item--divider:last-child {
  border-bottom: none;
}

/* Horizontal List */
.smolitux-list--horizontal {
  display: flex;
  flex-wrap: wrap;
}

.smolitux-list--horizontal .smolitux-list-item--horizontal {
  margin-right: 1rem;
}

/* Selectable List */
.smolitux-list--selectable .smolitux-list-item--clickable {
  cursor: pointer;
  transition: background-color 0.2s ease;
}

.smolitux-list--selectable .smolitux-list-item--clickable:hover {
  background-color: var(--list-item-hover-bg, #f3f4f6);
}

.smolitux-list--selectable .smolitux-list-item--selected {
  background-color: var(--list-item-selected-bg, #eff6ff);
  color: var(--list-item-selected-color, #1d4ed8);
}

/* List Item */
.smolitux-list-item {
  position: relative;
  display: flex;
  align-items: center;
  width: 100%;
}

.smolitux-list-item--disabled {
  opacity: 0.5;
  cursor: not-allowed;
  pointer-events: none;
}

.smolitux-list-item--clickable {
  cursor: pointer;
}

/* List Item Content */
.smolitux-list-item-content {
  flex: 1;
  min-width: 0;
}

.smolitux-list-item-primary {
  font-weight: 500;
  color: var(--list-item-primary-color, #1f2937);
}

.smolitux-list-item-secondary {
  font-size: 0.875em;
  color: var(--list-item-secondary-color, #6b7280);
  margin-top: 0.25rem;
}

/* List Item Icon */
.smolitux-list-item-icon {
  display: flex;
  align-items: center;
  justify-content: center;
  margin-right: 0.75rem;
  flex-shrink: 0;
  color: var(--list-item-icon-color, #6b7280);
}

/* List Item Action */
.smolitux-list-item-action {
  display: flex;
  align-items: center;
  margin-left: 0.75rem;
  flex-shrink: 0;
}

/* Description List */
.smolitux-list-item-term {
  font-weight: 500;
  color: var(--list-item-term-color, #1f2937);
}

.smolitux-list-item-description {
  color: var(--list-item-description-color, #6b7280);
  margin: 0;
}

/* Dark Mode */
.dark .smolitux-list--dividers .smolitux-list-item--divider {
  border-bottom-color: var(--list-divider-color-dark, #374151);
}

.dark .smolitux-list--selectable .smolitux-list-item--clickable:hover {
  background-color: var(--list-item-hover-bg-dark, #374151);
}

.dark .smolitux-list--selectable .smolitux-list-item--selected {
  background-color: var(--list-item-selected-bg-dark, #1e3a8a);
  color: var(--list-item-selected-color-dark, #93c5fd);
}

.dark .smolitux-list-item-primary {
  color: var(--list-item-primary-color-dark, #f3f4f6);
}

.dark .smolitux-list-item-secondary {
  color: var(--list-item-secondary-color-dark, #9ca3af);
}

.dark .smolitux-list-item-icon {
  color: var(--list-item-icon-color-dark, #9ca3af);
}

.dark .smolitux-list-item-term {
  color: var(--list-item-term-color-dark, #f3f4f6);
}

.dark .smolitux-list-item-description {
  color: var(--list-item-description-color-dark, #9ca3af);
}