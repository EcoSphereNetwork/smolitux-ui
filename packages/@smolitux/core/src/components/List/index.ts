// packages/@smolitux/core/src/components/List/index.ts
export { 
  default as List,
  ListItem,
  ListItemText,
  ListItemIcon,
  ListItemAction
} from './List';

export type { 
  ListProps,
  ListItemProps,
  ListItemTextProps,
  ListItemIconProps,
  ListItemActionProps,
  ListContextType
} from './List';