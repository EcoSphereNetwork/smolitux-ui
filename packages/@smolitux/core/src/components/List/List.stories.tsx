import type { Meta, StoryObj } from '@storybook/react';
import { List } from './List';

const meta: Meta<typeof List> = {
  title: 'Components/List',
  component: List,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
};

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    children: 'List',
  },
};

export const CustomStyle: Story = {
  args: {
    children: 'Custom List',
    className: 'custom-style',
  },
};

export const Interactive: Story = {
  args: {
    children: 'Interactive List',
    onClick: () => alert('Clicked!'),
  },
};
