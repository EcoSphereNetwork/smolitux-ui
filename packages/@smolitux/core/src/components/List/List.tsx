// packages/@smolitux/core/src/components/List/List.tsx
import React, { forwardRef } from 'react';
import './List.css';

export interface ListProps extends React.HTMLAttributes<HTMLUListElement> {
  /** Variante der Liste */
  variant?: 'default' | 'ordered' | 'unordered' | 'description' | 'custom';
  /** Größe der Liste */
  size?: 'sm' | 'md' | 'lg';
  /** Dichte der Liste */
  density?: 'compact' | 'default' | 'comfortable';
  /** Ausrichtung der Listenelemente */
  align?: 'start' | 'center' | 'end';
  /** Trennlinien zwischen Listenelementen anzeigen */
  dividers?: boolean;
  /** Horizontale Liste */
  horizontal?: boolean;
  /** Benutzerdefiniertes Icon für Listenelemente */
  icon?: React.ReactNode;
  /** Benutzerdefiniertes Marker-Symbol für ungeordnete Listen */
  marker?: 'disc' | 'circle' | 'square' | 'none' | string;
  /** Einrückung der Liste */
  indent?: boolean;
  /** Listenelemente können ausgewählt werden */
  selectable?: boolean;
  /** Ausgewähltes Listenelement (kontrollierter Modus) */
  selectedItem?: string | number | null;
  /** Callback bei Auswahl eines Listenelements */
  onSelectItem?: (id: string | number) => void;
}

export interface ListItemProps extends React.LiHTMLAttributes<HTMLLIElement> {
  /** Eindeutige ID des Listenelements */
  id?: string | number;
  /** Primärer Text */
  primary?: React.ReactNode;
  /** Sekundärer Text */
  secondary?: React.ReactNode;
  /** Icon links vom Text */
  icon?: React.ReactNode;
  /** Element rechts vom Text */
  action?: React.ReactNode;
  /** Ist das Element ausgewählt? */
  selected?: boolean;
  /** Ist das Element deaktiviert? */
  disabled?: boolean;
  /** Callback bei Klick auf das Listenelement */
  onClick?: (event: React.MouseEvent<HTMLLIElement>) => void;
}

export interface ListItemTextProps extends React.HTMLAttributes<HTMLDivElement> {
  /** Primärer Text */
  primary: React.ReactNode;
  /** Sekundärer Text */
  secondary?: React.ReactNode;
}

export interface ListItemIconProps extends React.HTMLAttributes<HTMLDivElement> {
  /** Icon-Element */
  children: React.ReactNode;
}

export interface ListItemActionProps extends React.HTMLAttributes<HTMLDivElement> {
  /** Aktions-Element */
  children: React.ReactNode;
}

export interface ListContextType {
  variant: ListProps['variant'];
  size: ListProps['size'];
  density: ListProps['density'];
  align: ListProps['align'];
  dividers: ListProps['dividers'];
  horizontal: ListProps['horizontal'];
  icon: ListProps['icon'];
  marker: ListProps['marker'];
  selectable: ListProps['selectable'];
  selectedItem: ListProps['selectedItem'];
  onSelectItem: ListProps['onSelectItem'];
}

export const ListContext = React.createContext<ListContextType>({
  variant: 'default',
  size: 'md',
  density: 'default',
  align: 'start',
  dividers: false,
  horizontal: false,
  icon: undefined,
  marker: undefined,
  selectable: false,
  selectedItem: null,
  onSelectItem: undefined,
});

/**
 * List-Komponente für die Anzeige von Daten in Listenform
 * 
 * @example
 * ```tsx
 * <List>
 *   <ListItem primary="Item 1" />
 *   <ListItem primary="Item 2" />
 * </List>
 * ```
 */
export const List = forwardRef<HTMLUListElement, ListProps>(({
  variant = 'default',
  size = 'md',
  density = 'default',
  align = 'start',
  dividers = false,
  horizontal = false,
  icon,
  marker,
  indent = true,
  selectable = false,
  selectedItem = null,
  onSelectItem,
  className = '',
  children,
  ...rest
}, ref) => {
  // Varianten-spezifische Klassen
  const variantClasses = {
    default: '',
    ordered: 'smolitux-list--ordered',
    unordered: 'smolitux-list--unordered',
    description: 'smolitux-list--description',
    custom: 'smolitux-list--custom'
  };

  // Größen-spezifische Klassen
  const sizeClasses = {
    sm: 'smolitux-list--sm',
    md: 'smolitux-list--md',
    lg: 'smolitux-list--lg'
  };

  // Dichte-spezifische Klassen
  const densityClasses = {
    compact: 'smolitux-list--compact',
    default: '',
    comfortable: 'smolitux-list--comfortable'
  };

  // Ausrichtungs-spezifische Klassen
  const alignClasses = {
    start: 'smolitux-list--align-start',
    center: 'smolitux-list--align-center',
    end: 'smolitux-list--align-end'
  };

  // Benutzerdefiniertes Marker-Symbol
  const markerStyle = marker && variant === 'unordered' ? {
    '--list-marker': typeof marker === 'string' && ['disc', 'circle', 'square', 'none'].includes(marker)
      ? marker
      : `"${marker}"`
  } as React.CSSProperties : undefined;

  // Zusammengesetzte Klassen
  const classes = [
    'smolitux-list',
    variantClasses[variant],
    sizeClasses[size],
    densityClasses[density],
    alignClasses[align],
    dividers && 'smolitux-list--dividers',
    horizontal && 'smolitux-list--horizontal',
    !indent && 'smolitux-list--no-indent',
    selectable && 'smolitux-list--selectable',
    className
  ].filter(Boolean).join(' ');

  // Element-Typ basierend auf der Variante
  const Component = variant === 'ordered' ? 'ol' : 'ul';

  return (
    <ListContext.Provider
      value={{
        variant,
        size,
        density,
        align,
        dividers,
        horizontal,
        icon,
        marker,
        selectable,
        selectedItem,
        onSelectItem
      }}
    >
      <Component
        ref={ref}
        className={classes}
        style={markerStyle}
        {...rest}
      >
        {children}
      </Component>
    </ListContext.Provider>
  );
});

/**
 * ListItem-Komponente für einzelne Listeneinträge
 * 
 * @example
 * ```tsx
 * <ListItem primary="Item 1" secondary="Beschreibung" />
 * ```
 */
export const ListItem = forwardRef<HTMLLIElement, ListItemProps>(({
  id,
  primary,
  secondary,
  icon: itemIcon,
  action,
  selected,
  disabled = false,
  onClick,
  className = '',
  children,
  ...rest
}, ref) => {
  const {
    variant,
    dividers,
    horizontal,
    icon: listIcon,
    selectable,
    selectedItem,
    onSelectItem
  } = React.useContext(ListContext);

  // Behandle Klick-Events
  const handleClick = (event: React.MouseEvent<HTMLLIElement>) => {
    if (disabled) return;
    
    if (selectable && id !== undefined && onSelectItem) {
      onSelectItem(id);
    }
    
    if (onClick) {
      onClick(event);
    }
  };

  // Bestimme, ob das Element ausgewählt ist
  const isSelected = selected !== undefined ? selected : (selectable && id !== undefined && id === selectedItem);

  // Zusammengesetzte Klassen
  const classes = [
    'smolitux-list-item',
    dividers && 'smolitux-list-item--divider',
    horizontal && 'smolitux-list-item--horizontal',
    isSelected && 'smolitux-list-item--selected',
    disabled && 'smolitux-list-item--disabled',
    (onClick || selectable) && !disabled && 'smolitux-list-item--clickable',
    className
  ].filter(Boolean).join(' ');

  // Verwende das Icon vom ListItem oder von der List
  const displayIcon = itemIcon || listIcon;

  // Render-Logik für verschiedene Varianten
  const renderContent = () => {
    // Wenn children vorhanden sind, verwende diese
    if (children) {
      return children;
    }

    // Für description-Listen
    if (variant === 'description') {
      return (
        <>
          {primary && <dt className="smolitux-list-item-term">{primary}</dt>}
          {secondary && <dd className="smolitux-list-item-description">{secondary}</dd>}
        </>
      );
    }

    // Standard-Darstellung
    return (
      <>
        {displayIcon && (
          <div className="smolitux-list-item-icon">
            {displayIcon}
          </div>
        )}
        <div className="smolitux-list-item-content">
          {primary && <div className="smolitux-list-item-primary">{primary}</div>}
          {secondary && <div className="smolitux-list-item-secondary">{secondary}</div>}
        </div>
        {action && (
          <div className="smolitux-list-item-action">
            {action}
          </div>
        )}
      </>
    );
  };

  // Für description-Listen
  if (variant === 'description') {
    return (
      <div ref={ref as React.Ref<HTMLDivElement>} className={classes} {...rest as any}>
        {renderContent()}
      </div>
    );
  }

  return (
    <li
      ref={ref}
      className={classes}
      onClick={handleClick}
      aria-disabled={disabled}
      aria-selected={isSelected}
      {...rest}
    >
      {renderContent()}
    </li>
  );
});

/**
 * ListItemText-Komponente für strukturierten Text in Listenelementen
 * 
 * @example
 * ```tsx
 * <ListItem>
 *   <ListItemText primary="Titel" secondary="Beschreibung" />
 * </ListItem>
 * ```
 */
export const ListItemText = forwardRef<HTMLDivElement, ListItemTextProps>(({
  primary,
  secondary,
  className = '',
  ...rest
}, ref) => {
  const classes = ['smolitux-list-item-content', className].filter(Boolean).join(' ');

  return (
    <div ref={ref} className={classes} {...rest}>
      {primary && <div className="smolitux-list-item-primary">{primary}</div>}
      {secondary && <div className="smolitux-list-item-secondary">{secondary}</div>}
    </div>
  );
});

/**
 * ListItemIcon-Komponente für Icons in Listenelementen
 * 
 * @example
 * ```tsx
 * <ListItem>
 *   <ListItemIcon><UserIcon /></ListItemIcon>
 *   <ListItemText primary="Benutzer" />
 * </ListItem>
 * ```
 */
export const ListItemIcon = forwardRef<HTMLDivElement, ListItemIconProps>(({
  children,
  className = '',
  ...rest
}, ref) => {
  const classes = ['smolitux-list-item-icon', className].filter(Boolean).join(' ');

  return (
    <div ref={ref} className={classes} {...rest}>
      {children}
    </div>
  );
});

/**
 * ListItemAction-Komponente für Aktionen in Listenelementen
 * 
 * @example
 * ```tsx
 * <ListItem>
 *   <ListItemText primary="Eintrag" />
 *   <ListItemAction><Button>Bearbeiten</Button></ListItemAction>
 * </ListItem>
 * ```
 */
export const ListItemAction = forwardRef<HTMLDivElement, ListItemActionProps>(({
  children,
  className = '',
  ...rest
}, ref) => {
  const classes = ['smolitux-list-item-action', className].filter(Boolean).join(' ');

  return (
    <div ref={ref} className={classes} {...rest}>
      {children}
    </div>
  );
});

List.displayName = 'List';
ListItem.displayName = 'ListItem';
ListItemText.displayName = 'ListItemText';
ListItemIcon.displayName = 'ListItemIcon';
ListItemAction.displayName = 'ListItemAction';

export default List;