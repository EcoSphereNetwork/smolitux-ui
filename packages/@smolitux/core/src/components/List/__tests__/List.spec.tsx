// packages/@smolitux/core/src/components/List/__tests__/List.spec.tsx
import React from 'react';
import { render } from '@testing-library/react';
import { List, ListItem, ListItemText, ListItemIcon, ListItemAction } from '../List';

describe('List Snapshot Tests', () => {
  it('matches snapshot with default props', () => {
    const { container } = render(
      <List>
        <ListItem primary="Item 1" />
        <ListItem primary="Item 2" />
      </List>
    );
    expect(container).toMatchSnapshot();
  });
  
  it('matches snapshot with ordered variant', () => {
    const { container } = render(
      <List variant="ordered">
        <ListItem primary="Item 1" />
        <ListItem primary="Item 2" />
      </List>
    );
    expect(container).toMatchSnapshot();
  });
  
  it('matches snapshot with unordered variant', () => {
    const { container } = render(
      <List variant="unordered">
        <ListItem primary="Item 1" />
        <ListItem primary="Item 2" />
      </List>
    );
    expect(container).toMatchSnapshot();
  });
  
  it('matches snapshot with description variant', () => {
    const { container } = render(
      <List variant="description">
        <ListItem primary="Term 1" secondary="Description 1" />
        <ListItem primary="Term 2" secondary="Description 2" />
      </List>
    );
    expect(container).toMatchSnapshot();
  });
  
  it('matches snapshot with dividers', () => {
    const { container } = render(
      <List dividers>
        <ListItem primary="Item 1" />
        <ListItem primary="Item 2" />
      </List>
    );
    expect(container).toMatchSnapshot();
  });
  
  it('matches snapshot with horizontal layout', () => {
    const { container } = render(
      <List horizontal>
        <ListItem primary="Item 1" />
        <ListItem primary="Item 2" />
      </List>
    );
    expect(container).toMatchSnapshot();
  });
  
  it('matches snapshot with selectable items', () => {
    const { container } = render(
      <List selectable selectedItem="item2">
        <ListItem id="item1" primary="Item 1" />
        <ListItem id="item2" primary="Item 2" />
      </List>
    );
    expect(container).toMatchSnapshot();
  });
  
  it('matches snapshot with icons and actions', () => {
    const { container } = render(
      <List>
        <ListItem>
          <ListItemIcon>
            <svg width="16" height="16" viewBox="0 0 16 16">
              <circle cx="8" cy="8" r="8" />
            </svg>
          </ListItemIcon>
          <ListItemText primary="Item with Icon and Action" />
          <ListItemAction>
            <button>Action</button>
          </ListItemAction>
        </ListItem>
      </List>
    );
    expect(container).toMatchSnapshot();
  });
  
  it('matches snapshot with different sizes', () => {
    const { container } = render(
      <>
        <List size="sm">
          <ListItem primary="Small Item" />
        </List>
        <List size="md">
          <ListItem primary="Medium Item" />
        </List>
        <List size="lg">
          <ListItem primary="Large Item" />
        </List>
      </>
    );
    expect(container).toMatchSnapshot();
  });
  
  it('matches snapshot with different densities', () => {
    const { container } = render(
      <>
        <List density="compact">
          <ListItem primary="Compact Item" />
        </List>
        <List density="default">
          <ListItem primary="Default Item" />
        </List>
        <List density="comfortable">
          <ListItem primary="Comfortable Item" />
        </List>
      </>
    );
    expect(container).toMatchSnapshot();
  });
});