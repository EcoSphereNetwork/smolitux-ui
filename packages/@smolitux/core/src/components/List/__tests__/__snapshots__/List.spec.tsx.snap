// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Snapshot Tests matches snapshot with default props 1`] = `
<div>
  <ul
    class="smolitux-list smolitux-list--md smolitux-list--align-start"
  >
    <li
      aria-disabled="false"
      aria-selected="false"
      class="smolitux-list-item"
    >
      <div
        class="smolitux-list-item-content"
      >
        <div
          class="smolitux-list-item-primary"
        >
          Item 1
        </div>
      </div>
    </li>
    <li
      aria-disabled="false"
      aria-selected="false"
      class="smolitux-list-item"
    >
      <div
        class="smolitux-list-item-content"
      >
        <div
          class="smolitux-list-item-primary"
        >
          Item 2
        </div>
      </div>
    </li>
  </ul>
</div>
`;

exports[`List Snapshot Tests matches snapshot with description variant 1`] = `
<div>
  <ul
    class="smolitux-list smolitux-list--description smolitux-list--md smolitux-list--align-start"
  >
    <div
      class="smolitux-list-item"
    >
      <dt
        class="smolitux-list-item-term"
      >
        Term 1
      </dt>
      <dd
        class="smolitux-list-item-description"
      >
        Description 1
      </dd>
    </div>
    <div
      class="smolitux-list-item"
    >
      <dt
        class="smolitux-list-item-term"
      >
        Term 2
      </dt>
      <dd
        class="smolitux-list-item-description"
      >
        Description 2
      </dd>
    </div>
  </ul>
</div>
`;

exports[`List Snapshot Tests matches snapshot with different densities 1`] = `
<div>
  <ul
    class="smolitux-list smolitux-list--md smolitux-list--compact smolitux-list--align-start"
  >
    <li
      aria-disabled="false"
      aria-selected="false"
      class="smolitux-list-item"
    >
      <div
        class="smolitux-list-item-content"
      >
        <div
          class="smolitux-list-item-primary"
        >
          Compact Item
        </div>
      </div>
    </li>
  </ul>
  <ul
    class="smolitux-list smolitux-list--md smolitux-list--align-start"
  >
    <li
      aria-disabled="false"
      aria-selected="false"
      class="smolitux-list-item"
    >
      <div
        class="smolitux-list-item-content"
      >
        <div
          class="smolitux-list-item-primary"
        >
          Default Item
        </div>
      </div>
    </li>
  </ul>
  <ul
    class="smolitux-list smolitux-list--md smolitux-list--comfortable smolitux-list--align-start"
  >
    <li
      aria-disabled="false"
      aria-selected="false"
      class="smolitux-list-item"
    >
      <div
        class="smolitux-list-item-content"
      >
        <div
          class="smolitux-list-item-primary"
        >
          Comfortable Item
        </div>
      </div>
    </li>
  </ul>
</div>
`;

exports[`List Snapshot Tests matches snapshot with different sizes 1`] = `
<div>
  <ul
    class="smolitux-list smolitux-list--sm smolitux-list--align-start"
  >
    <li
      aria-disabled="false"
      aria-selected="false"
      class="smolitux-list-item"
    >
      <div
        class="smolitux-list-item-content"
      >
        <div
          class="smolitux-list-item-primary"
        >
          Small Item
        </div>
      </div>
    </li>
  </ul>
  <ul
    class="smolitux-list smolitux-list--md smolitux-list--align-start"
  >
    <li
      aria-disabled="false"
      aria-selected="false"
      class="smolitux-list-item"
    >
      <div
        class="smolitux-list-item-content"
      >
        <div
          class="smolitux-list-item-primary"
        >
          Medium Item
        </div>
      </div>
    </li>
  </ul>
  <ul
    class="smolitux-list smolitux-list--lg smolitux-list--align-start"
  >
    <li
      aria-disabled="false"
      aria-selected="false"
      class="smolitux-list-item"
    >
      <div
        class="smolitux-list-item-content"
      >
        <div
          class="smolitux-list-item-primary"
        >
          Large Item
        </div>
      </div>
    </li>
  </ul>
</div>
`;

exports[`List Snapshot Tests matches snapshot with dividers 1`] = `
<div>
  <ul
    class="smolitux-list smolitux-list--md smolitux-list--align-start smolitux-list--dividers"
  >
    <li
      aria-disabled="false"
      aria-selected="false"
      class="smolitux-list-item smolitux-list-item--divider"
    >
      <div
        class="smolitux-list-item-content"
      >
        <div
          class="smolitux-list-item-primary"
        >
          Item 1
        </div>
      </div>
    </li>
    <li
      aria-disabled="false"
      aria-selected="false"
      class="smolitux-list-item smolitux-list-item--divider"
    >
      <div
        class="smolitux-list-item-content"
      >
        <div
          class="smolitux-list-item-primary"
        >
          Item 2
        </div>
      </div>
    </li>
  </ul>
</div>
`;

exports[`List Snapshot Tests matches snapshot with horizontal layout 1`] = `
<div>
  <ul
    class="smolitux-list smolitux-list--md smolitux-list--align-start smolitux-list--horizontal"
  >
    <li
      aria-disabled="false"
      aria-selected="false"
      class="smolitux-list-item smolitux-list-item--horizontal"
    >
      <div
        class="smolitux-list-item-content"
      >
        <div
          class="smolitux-list-item-primary"
        >
          Item 1
        </div>
      </div>
    </li>
    <li
      aria-disabled="false"
      aria-selected="false"
      class="smolitux-list-item smolitux-list-item--horizontal"
    >
      <div
        class="smolitux-list-item-content"
      >
        <div
          class="smolitux-list-item-primary"
        >
          Item 2
        </div>
      </div>
    </li>
  </ul>
</div>
`;

exports[`List Snapshot Tests matches snapshot with icons and actions 1`] = `
<div>
  <ul
    class="smolitux-list smolitux-list--md smolitux-list--align-start"
  >
    <li
      aria-disabled="false"
      aria-selected="false"
      class="smolitux-list-item"
    >
      <div
        class="smolitux-list-item-icon"
      >
        <svg
          height="16"
          viewBox="0 0 16 16"
          width="16"
        >
          <circle
            cx="8"
            cy="8"
            r="8"
          />
        </svg>
      </div>
      <div
        class="smolitux-list-item-content"
      >
        <div
          class="smolitux-list-item-primary"
        >
          Item with Icon and Action
        </div>
      </div>
      <div
        class="smolitux-list-item-action"
      >
        <button>
          Action
        </button>
      </div>
    </li>
  </ul>
</div>
`;

exports[`List Snapshot Tests matches snapshot with ordered variant 1`] = `
<div>
  <ol
    class="smolitux-list smolitux-list--ordered smolitux-list--md smolitux-list--align-start"
  >
    <li
      aria-disabled="false"
      aria-selected="false"
      class="smolitux-list-item"
    >
      <div
        class="smolitux-list-item-content"
      >
        <div
          class="smolitux-list-item-primary"
        >
          Item 1
        </div>
      </div>
    </li>
    <li
      aria-disabled="false"
      aria-selected="false"
      class="smolitux-list-item"
    >
      <div
        class="smolitux-list-item-content"
      >
        <div
          class="smolitux-list-item-primary"
        >
          Item 2
        </div>
      </div>
    </li>
  </ol>
</div>
`;

exports[`List Snapshot Tests matches snapshot with selectable items 1`] = `
<div>
  <ul
    class="smolitux-list smolitux-list--md smolitux-list--align-start smolitux-list--selectable"
  >
    <li
      aria-disabled="false"
      aria-selected="false"
      class="smolitux-list-item smolitux-list-item--clickable"
    >
      <div
        class="smolitux-list-item-content"
      >
        <div
          class="smolitux-list-item-primary"
        >
          Item 1
        </div>
      </div>
    </li>
    <li
      aria-disabled="false"
      aria-selected="true"
      class="smolitux-list-item smolitux-list-item--selected smolitux-list-item--clickable"
    >
      <div
        class="smolitux-list-item-content"
      >
        <div
          class="smolitux-list-item-primary"
        >
          Item 2
        </div>
      </div>
    </li>
  </ul>
</div>
`;

exports[`List Snapshot Tests matches snapshot with unordered variant 1`] = `
<div>
  <ul
    class="smolitux-list smolitux-list--unordered smolitux-list--md smolitux-list--align-start"
  >
    <li
      aria-disabled="false"
      aria-selected="false"
      class="smolitux-list-item"
    >
      <div
        class="smolitux-list-item-content"
      >
        <div
          class="smolitux-list-item-primary"
        >
          Item 1
        </div>
      </div>
    </li>
    <li
      aria-disabled="false"
      aria-selected="false"
      class="smolitux-list-item"
    >
      <div
        class="smolitux-list-item-content"
      >
        <div
          class="smolitux-list-item-primary"
        >
          Item 2
        </div>
      </div>
    </li>
  </ul>
</div>
`;
